path.c

```diff
@@ -1057,8 +1063,12 @@ void wglob_scandir_entry(
                     continue;
                 if (i + 1 < s->pattern.length) // has a next component?
                     t2->active_components[i + 1] = 1;
-                else
-                    wglob_add_result(s, only_if_existing, s->flags & WGLB_MARK);
+                else {
+                    char *bn = basename(s->path.contents);
+                    if (strcmp(bn, "..") != 0 && strcmp(bn, ".") != 0) {
```

Comment:
I think we could use `name` for the current component name, without the need to use `basename`:

```suggestion
                    if (strcmp(name, "..") != 0 && strcmp(name, ".") != 0) {
```

Reply:
It would be best to check `name` against `.` and `..` before the `xfnm_match` call. Doing so would bypass the unneeded pattern matching and also make the filtering effective for the `(i + 1 < s->pattern.length)` case.
For instance, `echo .*/tests` should not produce `./tests`.

Reply:
replaced `basename` with `name`, and moved the check above the `xfnm_match` call (hopefully this is what you meant?)

Reply:
Yes, thanks.